bin/dlang/codecov

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+# Copyright sociomantic labs GmbH 2017.
+# Distributed under the Boost Software License, Version 1.0.
+# (See accompanying file LICENSE.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+#
+# Build D applications
+set -eu
+
+# Paths
+r=$(dirname $0)/../..
+beaver=$r/bin/beaver
+
+# Use general dlang utilities
+. $r/lib/dlang.sh
+
+# Set the DC and DVER environment variables and export them to docker
+set_dc_dver
+
+# Export any codecov-specific environment variable
+codecov_vars="$(printenv -0 | sed -zn 's/^\(CODECOV_[^=]\+\)=.*$/\1\n/p' |
+        tr -d '\0')"
+# Export D related environment variables
+dlang_vars="DIST DMD DC DVER D2_ONLY F V"
```

Comment:
Shouldn't be needed at all, codecov script only works with generated .lst file and git history

Reply:
It's metadata, and I think is nice to know how the run was built to come up with that coverage. You can see it here for example (codecov sends `TRAVIS_OS` by default): https://codecov.io/gh/sociomantic-tsunami/ocean/commit/73ee571debc6e399e7acc465fc6dd28f9afe2cd7/build

We could also use labels for some things, like `F` and `DIST` could be good candidates, maybe even `DC`. (I don't know how they look, so I don't know how useful they would be visually). I might go for a try...

Reply:
OK, if you intend to submit it as part of metadata some way, it makes sense to me.

Reply:
Yup, it's done, have a look at ocean's PR :)